src/models/timeaxisheaderview.h

```diff
@@ -0,0 +1,86 @@
+#ifndef TIMEAXISHEADERVIEW_H
+#define TIMEAXISHEADERVIEW_H
+
+#include <QHeaderView>
+
+#include "data.h"
+
+class FilterAndZoomStack;
+
+class TimeAxisHeaderView : public QHeaderView
+{
+    Q_OBJECT
+public:
+    explicit TimeAxisHeaderView(FilterAndZoomStack* filterAndZoomStack, Qt::Orientation orientation, QWidget *parent = nullptr);
+
+public:
+    void setTimeRange(const Data::TimeRange &timeRange);
+private:
+    Data::TimeRange m_timeRange;
+
+private:
+    FilterAndZoomStack* m_filterAndZoomStack = nullptr;
+public slots:
+    void onFilterChanged(const Data::FilterAction &filterAction);
+    void onZoomChanged(const Data::ZoomAction &zoomAction);
+protected:
+    void paintSection(QPainter *painter, const QRect &rect, int logicalIndex) const override;
+};
+
+class PrefixTickLabels
```

Comment:
this class is private to the header view, no? so please move the declaration of this class either into an anon namespace within the .cpp file, or alternatively move it to a separate file .h/.cpp file pair